packages/frontend/src/hooks/useCompiledSql.ts

```diff
@@ -77,3 +77,19 @@ export const useCompiledSql = (
         ...queryOptions,
     });
 };
+
+export const useCompiledSqlFromMetricQuery = ({
+    tableName,
+    projectUuid,
+    metricQuery,
+}: {
+    tableName: string;
+    projectUuid: string;
+    metricQuery: MetricQuery;
+}) => {
+    return useQuery<ApiCompiledQueryResults, ApiError>({
+        queryKey: ['compiledQuery', tableName, metricQuery, projectUuid],
```

Comment:
I think this should be {projectUuid} / "compiledQuery" / {tableName} / {MetricQuery}

so if you invalidate project, this also gets invalidated

Reply:
@joaoviana I meant that when queryClient.invalidateQueries(['project', uuid]); runs somewhere, it won’t invalidate this query. Unfortunately, we don’t have a clear guide on how to structure query keys..